### PR TITLE
feat: integrate MoC oracle

### PR DESCRIPTION
**Pull request by kurt-wagner**

### CODE REVIEW

File: solidity/contracts/utility/MocBTCToUSDOracle.sol [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        return int256(value);
+    }
+
+    /**
+      * @dev returns the USD/BTC update time.
+      *
+      * @return always returns current block's timestamp
+    */
+    function latestTimestamp() external view returns (uint256) {
+        return now; // MoC oracle doesn't return update timestamp
```
Comment (kurt-wagner) [inline solidity/contracts/utility/MocBTCToUSDOracle.sol:42]: @ororopickpocket what do you think about this? Could this be a problem in the future?

Comment (ororopickpocket) [reply solidity/contracts/utility/MocBTCToUSDOracle.sol:42]: We don't have much of a choice here if MoC isn't returning the timestamp...
Where and for what is latestTimestamp() used by the protocol?

Comment (kurt-wagner) [reply solidity/contracts/utility/MocBTCToUSDOracle.sol:42]: It is used in the PriceOracle to get the last updated price timestamp

Comment (ororopickpocket) [reply solidity/contracts/utility/MocBTCToUSDOracle.sol:42]: update: the AMM is designed to use the internal rate if the price hasn't been updated more recently. With the changes introduced in this PR, we're always using the rate returned by the oracle for weight balancing. According to MoC the price is updated every 10 minutes or every time the price moves 0.1%, so it should be ok.

**Pull request merged**